frame/composable-traits/src/auction.rs

```diff
@@ -20,97 +17,17 @@ impl Default for AuctionStepFunction {
 	}
 }
 
-#[derive(Decode, Encode, Clone, PartialEq, TypeInfo)]
-pub enum AuctionState<DexOrderId> {
-	AuctionStarted,
-	AuctionOnDex(DexOrderId),
-	AuctionEndedSuccessfully,
-	/// like DEX does not support asset now or halted
-	AuctionFatalFailed,
-	/// so if for some reason system loop is not properly set, than will get timeout
-	AuctionTimeFailed,
-}
-
-impl<DexOrderId> Default for AuctionState<DexOrderId> {
-	fn default() -> Self {
-		Self::AuctionStarted
-	}
-}
-
-/// Auction is done via dexes which act each block. Each block decide if intention was satisfied or
-/// not. That information is provided via event subscribes which callback into auction.
-/// Assuming liquidity providers to be off our local chain, it means that it is high latency
-/// external loop.
-pub enum AuctionExchangeCallback {
-	/// success transfer of funds
-	Success,
-	/// some technical fail of transaction, can issue new one
-	RetryFail,
-	/// cannot retry within current state of system, like assets are not supported
-	FatalFail,
-}
-
-#[derive(Default, Decode, Encode, Clone, TypeInfo)]
+#[derive(Default, Decode, Encode, Clone, TypeInfo, Debug, PartialEq)]
 pub struct LinearDecrease {
-	/// The number of seconds until the price reach zero.
+	/// Seconds after auction start when the price reaches zero
 	pub total: DurationSeconds,
```

Comment:
Should we express this in blocknumbers, as seconds are less reliable.

Reply:
i do not know. i know that we use seconds in lending extensively. also I assume that seconds are synced in consensus. 

how predictable blocks spans are. if all blocks are exact 6 seconds. then there is no difference. 

if block can 5 or 7, or other kind of random, or consistenly skewed to 6.5 or runtime upgraded changes block time. than seconds are more "reliable" in terms of predictability of total auction duration i believe. so need to look into impl of Time pallet - it may be oracle with voting:) 

seconds are more intuitive, but for people it harder to target specific block number of sell with them. seconds are reliable for change in block time. but if block times are veryyy precies and exact, than second are not better.

Reply:
Then we'll keep it in seconds; I don't think it affects security.